graylog2-server/src/main/java/org/graylog2/indexer/indices/Indices.java

```diff
@@ -298,15 +298,29 @@ public long numberOfMessages(String indexName) throws IndexNotFoundException {
 
     public boolean exists(String indexName) {
         try {
-            return jestClient.execute(new IndicesExists.Builder(indexName).build()).isSucceeded();
+            final JestResult result = jestClient.execute(new GetSettings.Builder().addIndex(indexName).build());
+            if (!result.isSucceeded()) {
+                return false;
+            }
+            return Optional.of(result.getJsonObject())
+                .map(GsonUtils::entrySetAsMap)
+                .map(map -> map.containsKey(indexName))
+                .orElse(false);
         } catch (IOException e) {
             throw new ElasticsearchException("Couldn't check existence of index " + indexName, e);
         }
     }
 
     public boolean aliasExists(String alias) {
```

Comment:
Please add a short Javadoc comment what the method is supposed to do (and what not).